### PR TITLE
Hybrid Pruning

### PR DESCRIPTION
**Pull request by varunagrawal**
Got hybrid pruning to work and this PR is the result of it!

### CODE REVIEW

**Comment by varunagrawal** [comment]
@dellaert FYI this PR gets the 1D robot example working extremely well!

File: gtsam/hybrid/HybridSmoother.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+    // Conditionals to remove from the bayes net
+    // since the conditional will be updated.
+    std::vector<HybridConditional::shared_ptr> conditionals_to_erase;
+
+    // New conditionals to add to the graph
+    gtsam::HybridBayesNet newConditionals;
+
+    // NOTE(Varun) Using a for-range loop doesn't work since some of the
+    // conditionals are invalid pointers
```
Comment (ProfFan) [inline gtsam/hybrid/HybridSmoother.cpp:73]: Interesting, I don't think this is possible in a Hybrid Bayes Net?

**Review by ProfFan — COMMENTED** [review]
One question

File: gtsam/hybrid/HybridSmoother.cpp (hunk 1 above)
Comment (varunagrawal) [reply gtsam/hybrid/HybridSmoother.cpp:73]: It is! I was able to get a simple example up and running with this.

Comment (varunagrawal) [reply gtsam/hybrid/HybridSmoother.cpp:73]: I have no other explanation why a for-range loop doesn't work, unfortunately...

**Review by ProfFan — APPROVED** [review]

**Pull request merged**